api/java/src/main/java/com/qbot/answeringservice/controller/AnsweringController.java

```diff
@@ -1,47 +1,82 @@
 
 @Controller
+@RequestMapping("/exchanges")
 public class AnsweringController {
+    private final Logger logger = LoggerFactory.getLogger(AnsweringController.class);
 
+    @Autowired
     private final AnsweringService answeringService;
 
     public AnsweringController(AnsweringService answeringService) {
         this.answeringService = answeringService;
     }
 
-    @CrossOrigin(origins = {"http://localhost:3000", "https://qbot-slak.onrender.com"})
+    @CrossOrigin(origins = { "http://localhost:3000", "https://qbot-slak.onrender.com" })
     @GetMapping("/all")
-    public ResponseEntity<AllExchanges> getAllExchanges() {
-        // TODO: introduce auth
-        // if (verified)
-        // {
-        return ResponseEntity.ok(answeringService.getAllExchanges());
-        // }
-        // return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+    public ResponseEntity<ExchangeCollection> getAllExchanges() {
+        try {
+            // TODO: introduce auth
+            // if (verified)
+            // {
+            return ResponseEntity.ok(answeringService.getAllExchanges());
+            // }
+            // return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        } catch (Exception e) {
+            logger.error("Server error: message: " + e.getMessage());
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @CrossOrigin(origins = { "http://localhost:3000", "https://qbot-slak.onrender.com" })
+    @GetMapping(path = "/{courseId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<ExchangeCollection> getByCourseId(@PathVariable("courseId") String courseId) {
+        try {
+            if (courseId != null) {
+                return ResponseEntity.ok(answeringService.getExchangesByCourseId(courseId));
+            } else {
+                return ResponseEntity.badRequest().build();
+            }
+        } catch (Exception e) {
+            logger.error("Server error: message: " + e.getMessage());
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
     }
 
-    @CrossOrigin(origins = {"http://localhost:3000", "https://qbot-slak.onrender.com"})
-    @PostMapping("/questions")
-    public ResponseEntity<Void> saveQuestionsAndAnswerExchanges(@RequestBody AllExchanges exchanges) {
-        // TODO: introduce auth
-        // if (verified)
-        // {
-        if (answeringService.saveExchanges(exchanges)) {
-            return ResponseEntity.ok().build();
-        } else {
-            return ResponseEntity.badRequest().build();
+    @CrossOrigin(origins = { "http://localhost:3000", "https://qbot-slak.onrender.com" })
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> saveQuestionAndAnswerExchanges(@RequestBody ExchangeCollection exchanges) {
```

Comment:
Create another method for saveExchanges - this one is meant to overwrite everything.  This way we keep backwards compatability.

Reply:
Is that use case still applicable if we're moving to multiple courses?
If there are going to be multiple Educators or Admins in the system, they probably shouldn't have access to overwrite everything from one endpoint